### PR TITLE
[feat]: [PL-26074]: Changing searchUserGroupFilter Operator

### PR DESCRIPTION
**Pull request by Y0709**
Changed Search User Group Filter operator and added its unit test

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34529)

### CODE REVIEW

**Review by DeepakPatankar — DISMISSED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34529)* status: 2 file(s) to review. 1 of 2 approvals obtained.

File: 400-rest/src/main/java/software/wings/resources/UserGroupResource.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ public RestResponse<PageResponse<UserGroup>> list(@BeanParam PageRequest<UserGro
     if (!StringUtils.isEmpty(searchTerm)) {
       SearchFilter searchFilter = SearchFilter.builder()
                                       .fieldName(UserGroupKeys.name)
-                                      .op(Operator.STARTS_WITH)
+                                      .op(Operator.CONTAINS)
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/resources/UserGroupResource.java:109]: dont do this - this will have huge db impact

**Review by reetika04 — DISMISSED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34529)* status: :shipit: complete! ✓

**Comment by Y0709** [comment]
do not merge yet

**Review by abhinav-harness — COMMENTED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34529)* status: Some of the checks are too old.

**Review by abhinav-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34529)* status: Some of the checks are too old.

**Review by abhinav-harness — DISMISSED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34529)* status: 2 file(s) to review.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34529)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34529)* status: :shipit: complete! ✓

**Pull request merged**